### ISSUE TITLE
Solve TiKV disk full problem in kvproto

### ISSUE THREAD

**Issue by tier-cap** [pingcap/kvproto#803]
## Feature Request
1. When TiKV is disk full, some specific operations should be allowed. When leader accepts a proposal, and also wants to be allowed by the other peers, then some flags should be added in messages.

**Comment by tisonkun** [comment]
I think this issue is resolved by #788 and #783?

**Comment by disksing** [comment]
> I think this issue is resolved by #788 and #783?

I think so. @tier-cap may I close it?

**Comment by tier-cap** [comment]
> > I think this issue is resolved by #788 and #783?
> 
> I think so. @tier-cap may I close it?

Sorry, but may be not.
Because, the adjustment on the raft msg inflight in the refill logic need to be fixed with the second threshold diskusage. Currently, it only deals with the first threshold diskusage.
It may be good to keep this issue open rather than open another issue?

**Comment by tisonkun** [comment]
@tier-cap do you mean you take this issue into 3 parts, and now we complete 2 parts. There is still one part remaining?

**Comment by tier-cap** [comment]
> @tier-cap do you mean you take this issue into 3 parts, and now we complete 2 parts. There is still one part remaining?

Not take into more parts, only because there are some bugs fixing.
Closed after tikv/tikv#10833 merged may be suitable.

**Comment by tier-cap** [comment]
> @tier-cap thanks for your information.
> 
> [tikv/tikv#10833](https://github.com/tikv/tikv/pull/10833) doesn't close this issue as well as it seems has nothing to do with further updates in kvproto repo. I'd prefer let [tikv/tikv#10833](https://github.com/tikv/tikv/pull/10833) close this issue or just close this one if there is no further changes required on kvproto.

Right, currently no need to update kvproto, and 10833 only do some updates related to this issue in TiKV only. 
OK to close this now.
